src/Raven.Server/Config/Categories/AiConfiguration.cs

```
@@ -0,0 +1,90 @@
+    [ConfigurationEntry("Ai.Embeddings.Generation.Task.RetryStrategy", ConfigurationEntryScope.ServerWideOrPerDatabase)]
+    public EmbeddingsGenerationRetryStrategy EmbeddingsGenerationTaskRetryStrategy { get; set; }
+
+    #endregion
+
+    #region Querying
+
+    #region Caching
+
+    [Description("Maximum number of embeddings generated from query terms during vector searches that can be stored in the embeddings cache collection in a single batch operation. " +
+                 "Caching these embeddings reduces redundant processing and improves retrieval efficiency for identical queries.")]
+    [DefaultValue(128)]
+    [ConfigurationEntry("Ai.Embeddings.Generation.Querying.Caching.MaxBatchSize", ConfigurationEntryScope.ServerWideOrPerDatabase)]
+    public int QueryEmbeddingsGenerationMaxCacheBatchSize { get; set; }
+    
+    #endregion
+
+    #region Batching
+
+    [Description("Time in milliseconds to wait for additional query embedding requests before sending the current batch to the AI provider. " +
+                 "Lower values decrease latency for query embedding generation but may reduce throughput.")]
+    [DefaultValue(200)]
+    [ConfigurationEntry("Ai.Embeddings.Generation.Querying.Batching.TimeoutInMs", ConfigurationEntryScope.ServerWideOrPerDatabase)]
+    public int QueryEmbeddingsBatchTimeout { get; set; }
+
+    [Description("Maximum number of query embedding requests to include in a single batch sent to the AI provider. Optimal values depend on the provider's rate limits and pricing model.")]
+    [DefaultValue(128)]
+    [ConfigurationEntry("Ai.Embeddings.Generation.Querying.Batching.MaxBatchSize", ConfigurationEntryScope.ServerWideOrPerDatabase)]
+    public int QueryEmbeddingsMaxBatchSize { get; set; }
+
+    [Description("Maximum number of retry attempts for failed query embedding generation requests before giving up. Retries are performed using an exponential backoff strategy.")]
+    [DefaultValue(3)]
```

For queries? Why wouldn't we just throw immediately?
There is a _user_ on the other side, and we can return an error directly.

Really important to surface those issues - otherwise "ravendb is slow" because they have low rate limits.

---

I agree, throw immediately